src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,58 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAddMovies: (newMovie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovies }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [sppiner, setSppiner] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const createMovie = (foundMovie: MovieData): Movie => {
+      const image = foundMovie.Poster === 'N/A'
+        ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+        : foundMovie.Poster;
+
+      return {
+        title: foundMovie.Title,
+        description: foundMovie.Plot,
+        imgUrl: image,
+        imdbUrl: `https://www.imdb.com/title/${foundMovie.imdbID}`,
+        imdbId: foundMovie.imdbID,
+      };
+    };
+
+    getMovie(query)
+      .then((response) => {
+        if ('Error' in response) {
+          setHasTitleError(true);
+        } else {
+          setMovie(createMovie(response));
+        }
+      })
+      .finally(() => setSppiner(false));
+  };
+
   return (
     <>
-      <form className="find-movie">
+      <form
+        className="find-movie"
+        onSubmit={(e) => {
```

Comment:
Don't use one-letter variables, it's hard to find them in the IDE and might be confusing for other developers